### PR TITLE
add doWhileM for ZSchedule

### PR DESCRIPTION
**Pull request by contrun**
closes https://github.com/zio/zio/issues/1184

### CODE REVIEW

**Comment by contrun** [comment]
Sure. I am quite new to zio, and I didn't find test case of doWhile, so I did not add test cases. I will try to add some.

**Comment by jdegoes** [comment]
@contrun Thank you, and if you need any help, just let me know!

**Comment by ghostdogpr** [comment]
@contrun can you run `sbt fmt` to format all your files?

**Comment by contrun** [comment]
@ghostdogpr done

**Comment by ghostdogpr** [comment]
@contrun OK, the linting is good now, but there's an issue with the Scaladoc generation: https://circleci.com/gh/zio/zio/9636?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link (seems to be a bracket issue)

**Comment by contrun** [comment]
@ghostdogpr I think the scaladoc generation passed. But there is one more timeout error in zio.RTSSpec

```
TIMEOUT: 60 seconds (TestRuntime.scala:18)
```
I did not touch anything in RTSSpec. Can you please manually rerun it for me?

File: core/shared/src/main/scala/zio/Schedule.scala [hunk 1]
```diff
@@ -71,12 +71,24 @@ object Schedule {
...
   /**
    * See [[[ZSchedule.doUntil[A](f:* ZSchedule.doUntil]]]
    */
   final def doUntil[A](f: A => Boolean): Schedule[A, A] =
     ZSchedule.doUntil(f)
 
+  /**
+   * See [[ZSchedule.doUntilM]]
+   */
+  final def doUntilM[A](f: A => UIO[Boolean]): Schedule[A, A] =
```
Comment (jdegoes) [inline core/shared/src/main/scala/zio/Schedule.scala]: The only comment I'd make is that, `U` comes before `W`, but that problem was there before, so 👍 

Thanks for this contribution!

**Pull request merged**